source/blog/2017-07-12-unprivileged-containers-with-bwrap-oci-and-bubblewrap.md

```diff
@@ -0,0 +1,184 @@
+---
+title: Unprivileged containers with bwrap-oci and bubblewrap
+author: giuseppe
+date: 2017-07-12 13:00 UTC
+tags: bwrap-oci, usernamespaces, bubblewrap, unprivileged, containers
+published: true
+comments: true
+---
+The introduction of user namespaces in the Linux kernel has opened the
+doors to running containers as not privileged users.
+
+The concept behind user namespaces is quite simple: UIDs and GIDs in
+an user namespace are converted to a different set in the parent
+namespace, so that an application can be tricked to think it is
+executed by root while instead a not privileged user is running it.
+User namespaces are not limited to altering  an application’s  UID/GID
+mappings, a user can keep capabilities in the new namespace and
+together with other namespaces perform privileged operations there
+that are unprivileged in the parent namespace.
```

Comment:
For example, an application with a new network namespace can create firewall rules that only affect its namespace.